core/src/main/scala/com/clairvoyant/restonomer/core/app/RestonomerContext.scala

```diff
@@ -30,21 +30,34 @@ class RestonomerContext(val restonomerContextDirectoryPath: String) {
 
   val configVariables: Map[String, String] = loadConfigVariables(CONFIG_VARIABLES_FILE_PATH)
 
-  val checkpoints: List[CheckpointConfig] = loadConfigsFromDirectory[CheckpointConfig](
-    configDirectoryPath = CHECKPOINTS_CONFIG_DIRECTORY_PATH,
-    configVariablesSubstitutor = ConfigVariablesSubstitutor(configVariables = configVariables)
-  )
-
-  def runCheckpoint(checkpointName: String): Unit =
-    checkpoints
-      .find(_.name == checkpointName) match {
-      case Some(checkpointConfig) =>
-        runCheckpoint(checkpointConfig)
-      case None =>
-        throw new RestonomerContextException(s"The checkpoint: $checkpointName does not exists.")
+  def runCheckpoint(checkpointName: String): Unit = {
+    val checkpoints: List[CheckpointConfig] = loadConfigsFromDirectory[CheckpointConfig](
```

Comment:
In Scala, its generally recommended to not to mention the data type unless and until very much required. For example, In class variables it is suggested to have that and even IDE gives us a warning for class variables if we miss to mention the data type. But otherwise, we can ignore for local variables.

Reply:
resolved